options.go

```diff
@@ -75,3 +81,14 @@ func WithDisableTouchOnHit[K comparable, V any]() Option[K, V] {
 		opts.disableTouchOnHit = true
 	})
 }
+
+// WithTotalCost sets the maximum cost the cache is allowed to use (e.g. the used memory).
+// The actual cost calculation for each inserted item happens by making use of the
+// callback CostCalcFunc.
+// If used together with WithCapacity, WithTotalCost overrules the maximum capacity.
+func WithTotalCost[K comparable, V any](s uint64, callback CostCalcFunc[K, V]) Option[K, V] {
```

Comment:
```suggestion
func WithMaxCost[K comparable, V any](s uint64, callback CostFunc[K, V]) Option[K, V] {
```
As you suggested :+1:

Reply:
renamed